frontend/src/Pages/UserPage.tsx

```diff
@@ -60,57 +49,64 @@ export const UserPage = observer(() => {
 
-                    <div className={styles.registered}>#{user.id}, зарегистрирован <DateComponent date={user.registered} />
-                        {user.active && <span className={styles.active} title={'Активно посещал сайт в эту неделю'}>, <span className={'i i-alive'}></span>&nbsp;активен</span>}
-                        {!user.active && <span className={styles.active} title={'В последнюю неделю не заходил на сайт или заходил недостаточно часто, чтобы считаться активным'}>, <span className={'i i-ghost'}></span>&nbsp;неактивен</span>}
+                            <RatingSwitch rating={rating} type='user' id={user.id} double={true} votingDisabled={!restrictions?.canVoteKarma} onVote={handleOnVote}/>
+                        </div>
                     </div>
+                    <div className={styles.name}>{user.name}</div>
                 </div>
 
                 <div className={styles.controls}>
                     <Link className={`${styles.control} ${isProfile ? styles.active : ''}`} to={base}>Профиль</Link>
                     <Link className={`${styles.control} ${isPosts ? styles.active : ''}`} to={base + '/posts'}>Посты</Link>
                     <Link className={`${styles.control} ${isComments ? styles.active : ''}`} to={base + '/comments'}>Комментарии</Link>
                     <Link className={`${styles.control} ${isKarma ? styles.active : ''}`} to={base + '/karma'}>Саморегуляция</Link>
-                    {isMyProfile && restrictions?.canInvite && <Link className={`${styles.control} ${isInvites ? styles.active : ''}`} to={'/profile/invites'}>Инвайты</Link>}
-
+                    <Link className={`${styles.control} ${isInvites ? styles.active : ''}`} to={base + '/invites'}>Инвайты</Link>
                 </div>
+
                 <div className={styles.userinfo}>
                     {isProfile && <>
-                        {profile.invitedBy && <div>
-                            Зарегистрирован по приглашению <Username user={profile.invitedBy} />
-                        </div>}
+                        <div className={styles.registered}>#{user.id},
+                            {profile.invitedBy && <>
+                                <a href={`/u/${profile.invitedBy.username}/invites/#${user.username}`} title={'Детальный контекст приглашения'}>приглашен</a>
+                                <Username user={profile.invitedBy} /></> || <span>зарегистрирован</span>}
+                             <DateComponent date={user.registered} />
+                        </div>
                         {profile.invites.length > 0 && <div>
-                            По его приглашениям зарегистрированы: {profile.invites.map((user, idx) => {
-                                return <Username key={idx} user={user}/>;
+                            Пригласил: {profile.invites.map((user, idx) => {
```

Comment:
is it possible to user user gender here for dynamic label?

Reply:
Done.